accepted/0000-hub-online-synchronization.md

```diff
@@ -0,0 +1,228 @@
+
+Pros:
+- Automatically update channel during the night with repo-sync/mgr-sync
+- No need to have specific code adaptations in case of schema changes. It will be handled any way in repo-sync changes
+- Rely on a stable specification: repositories metadata
+- After initial configuration, all synchronization between hub and peripheral should be automatic and transparent
+  - Only differences would be transferred
+- Promote environments in CLM will automatically propagate changes to all connected peripherals
+- Automatic creating of bootstrap repository done by repo-sync tool
+
+Cons:
+- All peripheral can start synchronizing at same time
+  - Can be problematic if we have several peripherals performing full synchronization at same time. However we can configure the peripherals to run repo-sync in different hours and spread the load
+  - This is only a problem in first sync, since subsequent sync only transfer differences
+
+
+## Tools scope
+
+With this solution we can change the scope of the existing tools. For ISSv1 this would be a full replacement, meaning this tool/feature can be removed in SUSE Manager 5.0.
+
+Issv2 will continue to exist but with a change in its scope to be the tool to synchronize content between fully disconnected environments, and not to use in HUB set-ups. However, since this first implementation is not considering images and configuration channels synchronization, we still need to use it in HUB for these scenarios (should be temporary and replace with implementation in HUB Online Synchronization).
+
+The new Hub synchronization would be focused on HUB online deployments to allow scale SUSE Manager infrastructure. The only focus of this solution would be to work in scale and make it easier to use.
+
+
+# Detailed design
+[design]: #detailed-design
+
+The solution focuses on re-using the repo-sync mechanism to synchronize channels from Hub server to peripheral servers. It will be described in several steps that aim to address how it will be used and how we can technically support it.
+
+The main focus is to develop the integration between HUB server and peripherals through API only and re-using existing API methods as much as possible. However, some new custom methods will be needed.
+
+Configuration should be pushed from the HUB to peripherals, to avoid the need of direct management of the peripherals.
+
+For now this solution is focused on software channels only, but can be extended to synchronize more data types.
+
+
+## Define connection between HUB and Peripherals
+
+We can follow a similar approach to what exists on ISSv1. On the hub side we can define multiple peripheral servers to connect to by providing the FQDN and an authentication token.
+On the peripheral side we also need to define the Hub server FQDN and an Authentication token.
```

Comment:
Hub will have peripheral and generate associated auth token, right?

Why do we need to provide peripheral FQDN? Wouldn't generic peripheral name (may be FQDN) and generated token be enough? Or do we approach this as username/pass scenario?

I am assuming that connection will be always from peripheral to the hub, right?

Reply:
It should be used for authentication from HUB to peripheral. 
Communication will be bi-directional. Some cases will be peripheral calling hub (like synchronize software channels and calling SCC endpoints) other cases will be HUB calling the peripheral API like creating channels, pushing configuration channels, etc)